### ISSUE TITLE
AWS Batch submit command should respect `aws.client.storageEncryption`

### ISSUE THREAD

**Issue by bentsherman** [nextflow-io/nextflow#2808]
The `aws.client.storageEncryption` config option is used to append `--sse AES256` to s3 commands in the task script, so that encrypted buckets can be used. However, this option is not applied to the submit command generated by AWS Batch:

https://github.com/nextflow-io/nextflow/blob/cd95e2976bb6f6bcee7b9b48a28ec167be2ddca1/plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy#L600-L607

As a result, the `.command.log` file will not be uploaded. This bug is minor because it does not prevent the workflow from completing otherwise, and the `.command.out` and `.command.err` logs are uploaded correctly.

Basically, the above function needs to append the `--sse` flag if the `storageEncryption` setting is provided.
